### PR TITLE
🦁 Fix issues with missing dapp images and dapp names

### LINKED ISSUE #20731: 🧮 DyDx exchange doesn't show the app name signing screen with wallet-connect
# Bug Report

| Design | Implementation |
|--------|--------|
| image | IMG_9530 |

## Problem 
- [x]  The dapp name is not visible 
- [x] The message at the bottom is not adapting to the theme `Sign only if you trust the dapp`

## Additional issue
The background color of the missing dapp name is gray, but it should be a shade of the prominent color of the logo of dapp. 
@briansztamfater also had this issue while working with swaps. Sync with him to we don't repeat the soltution.

#### Expected behavior
- The dapp name should show up
- The info message at the bottom should respect the theme

### Reproduction

1. Open app
2. Open global scanner
3. Goto dydx exchange
4. Connect via wallet connect

### LINKED ISSUE #20732: 📷 DyDx missing profile photo on connection screen
# Bug Report

The connection screen of WalletConnect has issues when connection to DyDx.

| Design | Implementation |
|--------|--------|
| image | IMG_9529 | 

## Problem 
- [x] Dapp logo not visible (likely an issue with DyDx as it is visible for many other dapps / related to #20731)
- [x] The info message box theme doesn't adapt to darkness
- [x] The actions buttons below have extra horizontal padding (was fixed by someone else already)

### Reproduction

1. Try to connect to dydx exchange

### LINKED ISSUE #20769: WalletConnect: missing icon/name on certain dApps
# Bug Report

### Missing icons

The icons of some dApps are not displayed on the first connect confirmation screen and the `Connected dApps` screen.

Screenshot 2024-07-15 at 15 50 24
Screenshot 2024-07-15 at 15 50 43
Screenshot 2024-07-15 at 15 52 35

DApps from the [list](https://www.notion.so/Testing-Wallet-Connect-interactions-0739f176856848999f0ea1671e5dfbe0) with missing icon:

https://app.mellow.finance/
https://dydx.trade/ (also mentioned in https://github.com/status-im/status-mobile/issues/20731)
https://zapper.xyz/
https://yearn.fi/

#### Expected behavior
The icons are displayed 

#### Actual behavior
No icons

### Missing name 

Currently reproducible only for yearn.fi

Screenshot 2024-07-18 at 15 33 16

### Reproduction

1. Open a DApp, e.g. https://app.mellow.finance/
2. Scan the QR code with the Status scanner to connect your wallet
3. Check the confirmation screen
4. Connect to the dApp
5. Reopen the app (known issue that currently you need to relogin to see your connected dApps)
6. Check the `Connected dApps` screen

---

### [Design decision](https://github.com/status-im/status-desktop/issues/15593#issuecomment-2236471549) for the edge case when the dapp doesn't define any of the following: name, description, URL

In case the name and picture are missing, we should use the URL instead of the title and the Wallet User avatar component to generate icon.

https://www.figma.com/design/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=114-7646&t=z70ji8Dxuvm80yAe-1

Screenshot 2024-07-18 at 15 36 49

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by shivekkhurana**
fixes: #20732 #20731 #20769 

Also fixes an issue in comment: https://github.com/status-im/status-mobile/pull/20693#issuecomment-2228684100

| Title and Body | Image |
| --- | --- |
| **Edit missing icons on various dapps on connection screen**<br>The image still feels a little weird because dydx is exposing transparent pngs. | Simulator Screenshot - iPhone 13 - 2024-07-22 at 13 55 05 |
| **If icon is not present, use initials avatar**<br>Also, if name is missing, it is computed from the url. | Simulator Screenshot - iPhone 13 - 2024-07-22 at 13 34 13 |
| **Apply name and image fixes in connected dapps list**<br>Also, make the dapp icons rounded to follow the designs. | Simulator Screenshot - iPhone 13 - 2024-07-22 at 14 05 13 |
| **Fix Dapp name and icon not showing on signing, transaction screen** | Simulator Screenshot - iPhone 13 - 2024-07-22 at 13 55 40 |
| **Fix dapp connection info box not respecting theme colors** | Simulator Screenshot - iPhone 13 - 2024-07-22 at 14 01 22 |

Status: Ready

### CODE REVIEW

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 1]
```diff
@@ -282,3 +282,9 @@
             :params     [pairing-topic]
             :on-success #(log/info "Wallet Connect session disconnected")
             :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]}))
+
+(comment
+  (rf/dispatch [:wallet-connect/on-scan-connection "wc:70cd10e85be08cc94e18e41debf7707ec325d053f8ebfb958b5a6f5e0e03619a@2?expiryTimestamp=1721391801&relay-protocol=irn&symKey=958f40384b51c8f3bfb713c861e0131aa4aff82035d921f80515b8a24ee6a62b"])
+
+  ;;
+  )
```
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs]: leftover

**Review by clauxx — APPROVED** [review]
Looks good!

**Comment by pavloburykh** [comment]
@shivekkhurana thanks for the PR. 

### ISSUE 1 App and Account names are not aligned on signing request screen (Android)

Reproducing on Android 12, Samsung Galaxy A52

Below is an example from https://trade.dydx.exchange/portfolio/overview?utm_source=dydx-website

**Steps:** 

1. Scan wallet connect QR code
2. Proceed with signing connection requests triggered by dApp
3. Pay attention at the title within requests modal

**Actual result:** App name is not aligned. Account name background is overlapping the above text of the title. 

photo_2024-07-22 19 04 49


**Expected result:** screenshot from IOS

photo_2024-07-22 19 06 10

**Comment by pavloburykh** [comment]
### ISSUE 2 Long dApp names are broken within request modal

Here is an example from OpenSea https://opensea.io/

photo_2024-07-22 19 38 44

**Comment by shivekkhurana** [comment]
*Issue 1*
We already have this logged separately and Alex is working on it https://github.com/orgs/status-im/projects/65/views/39?sliceBy%5Bvalue%5D=2.30.0+Beta&pane=issue&itemId=70985293



*Issue 2*
I have asked Alisher for design directions and will act as soon as he responds.

**Comment by pavloburykh** [comment]
@shivekkhurana thanks for the PR. Except ISSUE 1 and 2 PR looks good to me. If those issues are being handled separately - PR is ready for merge.

**Pull request merged**